lib/errors.js

```diff
@@ -28,140 +28,135 @@ var BaseError = Base.inherits(Error, {
+    }
+}
+
+class InvalidPathError extends BaseError {
+    constructor(info) {
+        super('InvalidPathError', 'JS-INVALID-PATH-ERROR', 400, info);
+    }
+}
+
+class TimeoutError extends BaseError {
+    constructor(info) {
+        super('TimeoutError', 'JS-TIMEOUT-ERROR', 408, info);
+    }
+}
 
 errors.unknownError = function(err) {
-    return new errors.UnknownError({err: err});
+    return new UnknownError({err: err});
 };
 
 errors.juttleError = function(err, bundle) {
-    return new errors.JuttleError({err: err, bundle: bundle});
+    return new JuttleError({err: err, bundle: bundle});
 };
 
 errors.bundleError = function(reason, bundle) {
-    return new errors.BundleError({reason: reason, bundle: bundle});
+    return new BundleError({reason: reason, bundle: bundle});
 };
 
 errors.jobNotFoundError = function(job_id) {
-    return new errors.JobNotFoundError({job_id: job_id});
+    return new JobNotFoundError({job_id: job_id});
 };
 
 errors.fileNotFoundError = function(path) {
-    return new errors.FileNotFoundError({path: path});
+    return new FileNotFoundError({path: path});
 };
 
 errors.fileAccessError = function(path) {
-    return new errors.FileAccessError({path: path});
+    return new FileAccessError({path: path});
 };
 
 errors.directoryNotFoundError = function(path) {
-    return new errors.DirectoryNotFoundError({path: path});
+    return new DirectoryNotFoundError({path: path});
 };
 
 errors.directoryAccessError = function(path) {
-    return new errors.DirectoryAccessError({path: path});
+    return new DirectoryAccessError({path: path});
 };
 
 errors.invalidPathError = function(path) {
-    return new errors.InvalidPathError({path: path});
+    return new InvalidPathError({path: path});
 };
 
 errors.timeoutError = function(timeout) {
-    return new errors.TimeoutError({timeout: timeout});
+    return new TimeoutError({timeout: timeout});
 };
 
 errors.is_juttle_service_error = function(err) {
-    return (err instanceof errors.UnknownError ||
-            err instanceof errors.JuttleError ||
-            err instanceof errors.BundleError ||
-            err instanceof errors.JobNotFoundError ||
-            err instanceof errors.FileNotFoundError ||
-            err instanceof errors.FileAccessError ||
-            err instanceof errors.DirectoryNotFoundError ||
-            err instanceof errors.DirectoryAccessError ||
-            err instanceof errors.InvalidPathError ||
-            err instanceof errors.TimeoutError);
+    return (err instanceof UnknownError ||
+            err instanceof JuttleError ||
+            err instanceof BundleError ||
+            err instanceof JobNotFoundError ||
+            err instanceof FileNotFoundError ||
+            err instanceof FileAccessError ||
+            err instanceof DirectoryNotFoundError ||
+            err instanceof DirectoryAccessError ||
+            err instanceof InvalidPathError ||
+            err instanceof TimeoutError);
```

Comment:
I think it could. I wasn't certain instanceof worked properly with extendable-base, but it certainly does for classes.